experiments/pipeline/examples/general_usage.md

```diff
@@ -0,0 +1,195 @@
+Basic `dlt` functionalities are imported with `import dlt`. Those functionalities are:
+1. ability to run the pipeline (which means extract->normalize->load for particular source(s) and destination) with `dlt.run`
+2. ability to configure the pipeline ie. provide alternative pipeline name, working directory, folders to import/export schema and various flags: `dlt.pipeline`
+3. ability to decorate sources (`dlt.source`) and resources (`dlt.resources`)
+4. ability to access secrets `dlt.secrets` and config values `dlt.config`
+
+## importing destinations
+We support a few built in destinations which may be imported as follows
+```python
+import dlt
+from dlt.destinations import bigquery
+from dlt.destinations import redshift
+```
+
+The imported modules may be directly passed to `run` or `pipeline` method. They can be also called to provide credentials and other settings explicitly (discouraged) ie. `bigquery(Service.credentials_from_file("service.json"))` will bind the credentials to the module.
+
+Destinations require `extras` to be installed, if that is not the case, an exception with user friendly message will tell how to do that.
+
+## importing sources
+We do not have any structure for the source repository so IDK. For `create pipeline` workflow the source is in the same script as `run` method so the problem does not exist now (?).
+
+In principle, however, the importable sources are extractor functions so they are imported like any other function.
+
+
+## default and explicitly configured pipelines
+When the `dlt` is imported a default pipeline is automatically created. That pipeline is configured via configuration providers (ie. `config.toml` or env variables - see [secrets_and_config.md](secrets_and_config.md)). If no configuration is present, default values will be used.
+
+1. the name of the pipeline, the name of default schema (if not overridden by the source extractor function) and the default dataset (in destination) are set to **current module name** which in 99% of cases is the name of executing python script
+2. the working directory of the pipeline will be **OS temporary folder/pipeline name**
+3. the logging level will be **INFO**
+4. all other configuration options won't be set or will have default values.
+
+Pipeline can be explicitly created and configured via `dlt.pipeline()` that returns `Pipeline` object. All parameters are optional. If no parameter is provided then default pipeline is returned. Here's a list of options. All the options are configurable.
+1. pipeline_name - default as above
+2. working_dir - default as above
+3. pipeline_secret - for deterministic hashing - default is random number
+4. destination - the imported destination module or module name (we accept strings so they can be configured) - default is None
+5. import_schema_path - default is None
+6. export_schema_path - default is None
+7. full_refresh - if set to True all the pipeline working dir and all datasets will be dropped with each run
```

Comment:
You are right. On the other hand, the development process without this flag is PITA. you typically start with pipeline that you want to destory with each run because a lot of things changes and you reload the data.

what is your advice? the option can be dropped. we can create a `dlt pipeline destroy` command that is always interactive and teach people how to use it etc.

my idea also was that every `dlt deploy` command would disable this option and it would not run in the container etc.